src/software/sensor_fusion/filter/possession_filter.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <vector>
+
+#include "shared/constants.h"
+#include "software/world/ball.h"
+#include "software/world/robot_state.h"
+#include "software/world/team.h"
+#include "software/world/timestamped_possession_state.h"
+
+/**
+ * Compiles the list of robots that have possession of the ball
+ *
+ * @param friendly_robots_with_breakbeam_triggered The friendly robots that have their
+ * breakbeam triggered
+ * @param friendly_team The friendly_team of robots
+ * @param enemy_team The enemy_team of robots
+ * @param ball The ball
+ *
+ * @return the list of robots that possession of the ball
+ */
+std::vector<RobotIdWithTeamSide> getRobotsWithPossession(
+    std::vector<RobotId> friendly_robots_with_breakbeam_triggered, Team friendly_team,
+    Team enemy_team, Ball ball);
+
+/**
+ * Decides if the ball is near the dribbler of the robot
```

Comment:
Do we want to provide a threshold for "near"? Otherwise maybe another name would make it more clear how close "near" is? eg. `ballInControlOfDribbler` or other names that I think were suggested before (from the original `ballInMouth` discussion)

Actually is this function really a part of possession if we assume robots have `ballInMouth`?

Reply:
`ballInMouth` is not a thing anymore. Here I'm using the definition of possession from evaluation functions + the robots with broken beams

Reply:
I thought we were going to add ballInMouth to the Robot class, just not the RobotState class?

Reply:
No, I thought the idea was that ballInMouth is too low level and SensorFusion should abstract it away as possession.